staging/src/k8s.io/legacy-cloud-providers/openstack/openstack_instances.go

```diff
@@ -48,6 +48,19 @@ const (
 func (os *OpenStack) Instances() (cloudprovider.Instances, bool) {
 	klog.V(4).Info("openstack.Instances() called")
 
+	// For Kubelet we don't have to initialize cloud provider as it
+	// can read all required data from metadata service. Absence of
+	// secretLister means that we use kubelet.
+	if os.secretLister == nil {
+		return &Instances{
```

Comment:
while testing this change - what I found is, kubelet still needs cloud-config even if it is invalid one. Can we make `cloud-config` an optional parameter for cloud-provider initialization so as kubelet does not crashes if no cloud-config is provided?

Reply:
Done! now `cloud-provider` is an optional parameter.